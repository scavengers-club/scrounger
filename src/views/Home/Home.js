import { useState, useEffect } from 'react';
import { getAdventures } from '../../services/adventures';
import AdventureList from '../../components/AdventureList/AdventureList';

export default function Home({ currentUser }) {
  const [adventures, setAdventures] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const resp = await getAdventures();
      setAdventures(resp);
      setLoading(false);
    };
    fetchData();
  }, []);

  if (loading) return <h1>grabbing adventures...</h1>;

  return (
    <div>
      <img className="logo" src={`${process.env.PUBLIC_URL}/logo.png`} />

      {!currentUser && (
        <p className="message-box">
          Welcome explorer! We&apos;re here to guide you on real-world adventures. Scrounger
          encourages you to explore, discover the world, and cultivate community by creating shared
          experiences.
        </p>
      )}
      {currentUser && <h3 className="message-box">Welcome back, {currentUser.username}!</h3>}
      <AdventureList {...{ adventures }} />
    </div>
  );
}
