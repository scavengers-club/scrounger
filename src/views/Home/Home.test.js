import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import Home from './Home';

test('<Home/> should welcome current user', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/']}>
      <Route path="/">
        <Home currentUser={{ id: '575e91dd-2c9c-4297-aa60-6504a62236dc', username: 'violet' }} />
      </Route>
    </MemoryRouter>
  );

  await screen.findByText('Welcome back, violet!');
  expect(container).toMatchSnapshot();
});
