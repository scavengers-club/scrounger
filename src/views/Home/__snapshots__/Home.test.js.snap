// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> should welcome current user 1`] = `
<div>
  <div>
    <img
      class="logo"
      src="/logo.png"
    />
    <h3
      class="message-box"
    >
      Welcome back, 
      violet
      !
    </h3>
    <div
      class="adventure-list"
    >
      <div
        class="adventure-card"
      >
        <a
          class="list-link"
          href="/adventure/29"
        >
          <h3
            class="list-title"
          >
            Wishing Tree
          </h3>
          <img
            alt="Wishing Tree"
            class="list-img"
            src="https://kfnfolzoauwbouvhqbrv.supabase.co/storage/v1/object/public/adv-images/wishing-tree/wishing-tree.jpg"
          />
        </a>
      </div>
      <div
        class="adventure-card"
      >
        <a
          class="list-link"
          href="/adventure/30"
        >
          <h3
            class="list-title"
          >
            Witch’s Castle
          </h3>
          <img
            alt="Witch’s Castle"
            class="list-img"
            src="http://pnwphotoblog.com/wp-content/uploads/sites/11/2010/05/Stone-House.jpg"
          />
        </a>
      </div>
      <div
        class="adventure-card"
      >
        <a
          class="list-link"
          href="/adventure/28"
        >
          <h3
            class="list-title"
          >
            Waterfront Docks
          </h3>
          <img
            alt="Waterfront Docks"
            class="list-img"
            src="https://kfnfolzoauwbouvhqbrv.supabase.co/storage/v1/object/public/adv-images/waterfront-docks/marine drive docks.jpeg"
          />
        </a>
      </div>
      <div
        class="adventure-card"
      >
        <a
          class="list-link"
          href="/adventure/16"
        >
          <h3
            class="list-title"
          >
            Portland House
          </h3>
          <img
            alt="Portland House"
            class="list-img"
            src="https://dynamic-media-cdn.tripadvisor.com/media/photo-o/14/10/2f/7b/portland.jpg?w=700&h=500&s=1"
          />
        </a>
      </div>
      <div
        class="adventure-card"
      >
        <a
          class="list-link"
          href="/adventure/2"
        >
          <h3
            class="list-title"
          >
            Old Portland Adventure
          </h3>
          <img
            alt="Old Portland Adventure"
            class="list-img"
            src="https://kfnfolzoauwbouvhqbrv.supabase.in/storage/v1/object/public/adv-images/streetsign.jpg"
          />
        </a>
      </div>
      <div
        class="adventure-card"
      >
        <a
          class="list-link"
          href="/adventure/1"
        >
          <h3
            class="list-title"
          >
            mount hood
          </h3>
          <img
            alt="mount hood"
            class="list-img"
            src="https://kfnfolzoauwbouvhqbrv.supabase.in/storage/v1/object/public/adv-images/trillium.jpg"
          />
        </a>
      </div>
      <div
        class="adventure-card"
      >
        <a
          class="list-link"
          href="/adventure/3"
        >
          <h3
            class="list-title"
          >
            street corner
          </h3>
          <img
            alt="street corner"
            class="list-img"
            src="https://kfnfolzoauwbouvhqbrv.supabase.in/storage/v1/object/public/adv-images/portlandcitypark.jpg"
          />
        </a>
      </div>
      <div
        class="adventure-card"
      >
        <a
          class="list-link"
          href="/adventure/15"
        >
          <h3
            class="list-title"
          >
            bridge magic
          </h3>
          <img
            alt="bridge magic"
            class="list-img"
            src="https://static.onecms.io/wp-content/uploads/sites/28/2021/05/06/portand-oregon-PORTLANDTG0521.jpg"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
